webapp/channels/src/components/admin_console/system_roles/system_roles.tsx

```diff
@@ -45,69 +45,70 @@ const columns: Column[] = [
+                    role: <FormattedMessage {...rolesStrings[role.name].name}/>,
+                    description: <FormattedMessage {...rolesStrings[role.name].description}/>,
+                    type: <FormattedMessage {...rolesStrings[role.name].type}/>,
+                    edit: (
+                        <span
+                            className='SystemRoles_editRow'
+                            data-testid={`${role.name}_edit`}
                         >
-                            <div className='SystemRoles'>
-                                <DataGrid
-                                    rows={rows}
-                                    columns={columns}
-                                    page={1}
-                                    startCount={0}
-                                    endCount={rows.length}
-                                    loading={false}
-                                    nextPage={() => {}}
-                                    previousPage={() => {}}
+                            <Link to={`/admin_console/user_management/system_roles/${role.id}`} >
+                                <FormattedMessage
+                                    id='admin.permissions.roles.edit'
+                                    defaultMessage='Edit'
                                 />
-                            </div>
-                        </AdminPanel>
-                    </div>
+                            </Link>
+                        </span>
+                    ),
+                },
+                onClick: () => getHistory().push(`/admin_console/user_management/system_roles/${role.id}`),
+            });
+        }
+    });
+
+    return (
+        <div className='wrapper--fixed'>
+            <AdminHeader>
+                <FormattedMessage
+                    id='admin.permissions.systemRoles'
+                    defaultMessage='Delegated Granular Administration'
+                />
+            </AdminHeader>
+            <div className='admin-console__wrapper'>
+                <div className='admin-console__content'>
+                    <AdminPanel
+                        id='SystemRoles'
+                        title={defineMessage({id: 'admin.permissions.systemRolesBannerTitle', defaultMessage: 'Admin Roles'})}
+                        subtitle={defineMessage({id: 'admin.permissions.systemRolesBannerText', defaultMessage: 'Manage different levels of access to the system console.'})}
+                    >
+                        <div className='SystemRoles'>
+                            <DataGrid
+                                rows={rows}
+                                columns={columns}
+                                page={1}
+                                startCount={0}
+                                endCount={rows.length}
+                                loading={false}
+                                nextPage={() => {}}
+                                previousPage={() => {}}
```

Comment:
Create a noop function outside of the component and pass it instead of the arrows functions here (so you don't create a new function on every render).